crates/base/src/datetimes.rs

```diff
@@ -49,13 +51,17 @@ pub struct BaseTimeZone {
     offset: i32,
 }
 
-/// The time zone ID is an `u16` which represents a variety of the `enum` `chrono_tz::Tz`.
+/// The time zone ID is an `i16` which represents:
 ///
-/// There are 594 variants in total, so `u16` here is needed to store the index.
+/// * for values `x` that `x >= 0`, the index of a variety of the `enum` `chrono_tz::Tz`. There
+///   are 594 variants in total, so the it needs 10 lower bits to store the index.
+/// * for values `x` that `x < 0`, its lower byte of it (an `i8`) represents the fixed time zone
+///   offset in quarters. E.g., `5` means `'+01:15'` and `-12` means `'-03:00'`, etc.
+///   Note, `'-00:00'` will be converted to `'+00:00'`.
 #[derive(
     Serialize, Deserialize, Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash,
 )]
-pub struct TimeZoneId(u16);
+pub struct TimeZoneId(i16);
```

Comment:
I make a thought: we just use TimeZoneId(i32). The performance impact is rare, in fact. We just use offset in seconds. We can discard TimeZoneId as well. That depends on your thinking.

Reply:
pub struct TimeZoneId(i16) -> pub struct TimeZoneId(pub i32)
TimeZoneId(0) is just UTC. We do not need Tz::UTC -> TimeZoneId